src/main/kotlin/no/nav/familie/ks/sak/barnehagelister/domene/BarnehagebarnRepository.kt

```diff
@@ -4,137 +4,53 @@ import org.springframework.data.domain.Page
-        dagensDato: LocalDate,
-        pageable: Pageable,
-    ): Page<BarnehagebarnDtoInterface>
-
-    @Query(
-        """
-            SELECT bb.ident as ident, bb.fom as fom, bb.tom as tom, bb.antall_timer_i_barnehage as antallTimerIBarnehage,
-            bb.endringstype as endringstype, bb.kommune_navn as kommuneNavn, bb.kommune_nr as kommuneNr, MAX(bb.endret_tid) as endretTid,
-            f.id as fagsakId, f.status as fagsakstatus
-            FROM barnehagebarn bb
-            LEFT OUTER JOIN personident p ON bb.ident = p.foedselsnummer
-            LEFT OUTER JOIN po_person pp ON p.fk_aktoer_id = pp.fk_aktoer_id
-            LEFT OUTER JOIN gr_personopplysninger go ON pp.fk_gr_personopplysninger_id = go.id
-            LEFT OUTER JOIN  behandling b ON go.fk_behandling_id = b.id AND b.aktiv = true
-            LEFT OUTER JOIN fagsak f ON b.fk_fagsak_id = f.id and f.arkivert = false 
-            WHERE UPPER(bb.kommune_navn) = UPPER(:kommuneNavn)
-            GROUP BY bb.ident, bb.fom, bb.tom, bb.antall_timer_i_barnehage, bb.endringstype, bb.kommune_navn, bb.kommune_nr, f.id, f.status""",
+        """WITH lopende_andel_i_aktiv_behandling_for_ident AS (SELECT p.foedselsnummer, aty.stonad_tom >= current_date as lopende_andel
+                                                                 FROM andel_tilkjent_ytelse aty
+                                                                          JOIN behandling b ON aty.fk_behandling_id = b.id
+                                                                          JOIN personident p ON aty.fk_aktoer_id = p.fk_aktoer_id
+                                                                 WHERE p.aktiv
+                                                                   AND b.aktiv),
+     avvik_antall_timer_vilkar_resultat_og_barnehagebarn AS (SELECT bb.id                                          as barnehagebarn_id,
+                                                                    vr.antall_timer != bb.antall_timer_i_barnehage as avvik
+                                                              FROM barnehagebarn bb
+                                                                       JOIN personident p ON bb.ident = p.foedselsnummer and p.aktiv
+                                                                       JOIN person_resultat pr ON pr.fk_aktoer_id = p.fk_aktoer_id
+                                                                       JOIN vilkar_resultat vr
+                                                                            ON pr.id = vr.fk_person_resultat_id
+                                                                                AND vr.vilkar = 'BARNEHAGEPLASS'
+                                                                                AND bb.fom = vr.periode_fom::date)
+SELECT bb.ident,
+       bb.fom,
+       bb.tom,
+       bb.antall_timer_i_barnehage as antallTimerBarnehage,
+       bb.endringstype             as endringstype,
```

Comment:
Mulig du ikke trenger denne 

```suggestion
       bb.endringstype,
```